cmd/jaeger/internal/exporters/storageexporter/exporter_test.go

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2024 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storageexporter
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/extension"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/cmd/jaeger/internal/extension/jaegerstorage"
+	memoryCfg "github.com/jaegertracing/jaeger/pkg/memory/config"
+)
+
+var (
+	storage_exporter *storageExporter
+	storage_host     storageHost
+)
+
+type storageHost struct {
+	component *component.Component
+}
+
+func (host storageHost) GetExtensions() map[component.ID]component.Component {
+	myMap := make(map[component.ID]component.Component)
+	myMap[component.NewID("jaeger_storage")] = *host.component
+	return myMap
+}
+
+func (storageHost) ReportFatalError(err error) {
+	fmt.Println(err)
+}
+
+func (storageHost) GetFactory(_ component.Kind, _ component.Type) component.Factory {
+	return nil
+}
+
+func (storageHost) GetExporters() map[component.DataType]map[component.ID]component.Component {
+	return nil
+}
+
+const memstoreName = "memstore"
+
+func TestExporter(t *testing.T) {
+	config := &Config{}
+	config.TraceStorage = memstoreName
+	telemetry_settings := component.TelemetrySettings{
+		Logger: zap.L(),
+	}
+	storage_exporter = newExporter(config, telemetry_settings)
+	assert.Equal(t, storage_exporter.logger, telemetry_settings.Logger)
+	assert.Equal(t, storage_exporter.config, config)
+
+	storage_factory := jaegerstorage.NewFactory()
+	storage_config := jaegerstorage.Config{Memory: map[string]memoryCfg.Configuration{
+		memstoreName: {MaxTraces: 10000},
+	}}
+	storage_config.Memory[memstoreName] = memoryCfg.Configuration{MaxTraces: 10000}
+	storage_component, _ := storage_factory.CreateExtension(
+		context.Background(),
+		extension.CreateSettings{
+			TelemetrySettings: telemetry_settings,
+		},
+		&storage_config)
+	storage_host = storageHost{
+		component: &storage_component,
+	}
+
+	storage_component.Start(context.Background(), storage_host)
+	err := storage_exporter.start(context.Background(), storage_host)
```

Comment:
Why are you using private method to start the exporter? The objective of the ticket is to create tests that provide 100% test coverage, you're currently at 60% only

Reply:
Sorry I did not follow. There are only private methods in the exporter. Is there any other way to start the exporter?

Reply:
The public methods are on the interfaces, NewFactory, factory.CreateTracesExporter, etc. Ideally all tests should be calling into public methods, to exercise contractually promised behavior, not private methods which are implementation details.

Reply:
Ok. I will make the necessary changes to the test then.